compiler/resolution/functionResolution.cpp

```diff
@@ -5596,6 +5597,7 @@ preFold(Expr* expr) {
       }
       call->replace(result);
     } else if (call->isPrimitive(PRIM_FIELD_ID_BY_NUM)) {
+      /* this is really for unions */
```

Comment:
I don't really understand this comment.  Is it saying that this primitive is only applicable to unions?  (checking the uses, this seems to be the case).  I guess "really" is the part that makes me stumble -- it suggests that there might be other cases in which one might (try to / be able to / want to) apply it.